### PR TITLE
Fusetools 2997 fix destination camel foldr button

### PR DESCRIPTION
**Pull request by apupier**
# Pull Request Checklist

After this checklist is all checked or PR provides explanations for possible pass-through, please put the label "Ready for review" on the PR.

## General

- [x] Did you use the Jira Issue number in the commit comments?
- [x] Did you set meaningful commit comments on each commit?
- [x] Did you sign off all commits for this PR? (git commit -s -m "jira issue number - your commit comment")

## Functional

- [x] Did the CI job report a successful build?
- [x] Is the non-happy path working, too?

## Maintainability

- [x] Are all Sonar reported issues fixed or can they be ignored?
- [x] Is there no duplicated code?
- [x] Are method-/class-/variable-names meaningful?

## Tests

- [ ] Are there unit-tests?
- [ ] Are there integration tests (or at least a jira to tackle these)?
- [ ] Do we need a new UI test?

## Legal

- [x] Have you used the correct file header copyright comment?
- [x] Have you used the correct year in the headers of new files?

### CODE REVIEW

**Comment by apupier** [comment]
wait for https://github.com/jbosstools/jbosstools-fuse/pull/1483 to be merged to review (or use the review by commit) https://github.com/jbosstools/jbosstools-fuse/pull/1484/commits/d2c1e6fe635dc5074a918712d5154a948993a2a1

it sounds more like a workaround than a proper implementation but not found the root cause issue :(

I think it makes more sense to have a UI test for that.

File: editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/pages/Wsdl2RestWizardSecondPage.java [hunk 1]
```diff
@@ -51,61 +48,75 @@ public void createControl(Composite parent) {
...
 			public void widgetSelected(SelectionEvent e) {
 				// browse
 				IProject project = ResourcesPlugin.getWorkspace().getRoot().getProject(getOptionsFromWizard().getProjectName());
 				String path = selectFolder(project);
 				if (path != null) {
-					getOptionsFromWizard().setDestinationJava(path);
-					javaPathTextControl.notifyListeners(SWT.Modify, new Event());
+					getOptionsFromWizard().setDestinationCamel(path);
+					camelPathTextControl.notifyListeners(SWT.Modify, new Event());
+					camelPathBinding.updateModelToTarget();
```
Comment (apupier) [inline editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/pages/Wsdl2RestWizardSecondPage.java]: here is the first real behavior modification (other parts are just refactoring)

File: editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/pages/Wsdl2RestWizardSecondPage.java [hunk 2]
```diff
@@ -51,61 +48,75 @@ public void createControl(Composite parent) {
...
 			public void widgetSelected(SelectionEvent e) {
 				// browse
 				IProject project = ResourcesPlugin.getWorkspace().getRoot().getProject(getOptionsFromWizard().getProjectName());
 				String path = selectFolder(project);
 				if (path != null) {
-					getOptionsFromWizard().setDestinationCamel(path);
-					camelPathTextControl.notifyListeners(SWT.Modify, new Event());
+					getOptionsFromWizard().setDestinationJava(path);
+					javaPathTextControl.notifyListeners(SWT.Modify, new Event());
+					javaPathBinding.updateModelToTarget();
```
Comment (apupier) [inline editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/pages/Wsdl2RestWizardSecondPage.java]: here is the second and last real behavior modification (other parts are just refactoring)

**Comment by bfitzpat** [comment]
weird build errors upstream?

**Comment by apupier** [comment]
> weird build errors upstream?

yes, due to unstability of product/tests/CI
relaunched the build

**Comment by apupier** [comment]
created issue to provide UI test https://issues.jboss.org/browse/FUSETOOLS-3156

**Pull request merged**